packages/Search/src/details/DTK_DetailsDistributor.hpp

```diff
@@ -0,0 +1,246 @@
+    //}, Kokkos::Experimental::Max<int, Kokkos::HostSpace>(next_biggest_rank));
+    next_biggest_rank = *std::max_element(tmp_ranks.data(), tmp_ranks.data() + n);
+    if (next_biggest_rank == -1) break;
+    unique_ranks.push_back(next_biggest_rank);
+    offsets.push_back(offset);
+    Kokkos::View<int, Kokkos::HostSpace> total("total");
+    Kokkos::parallel_scan("process biggest rank items", Kokkos::RangePolicy<Kokkos::Serial>(0, n),
+    KOKKOS_LAMBDA(ST i, int& index, const bool last_pass) {
+      if (last_pass && (tmp_ranks(i) == next_biggest_rank)) {
+        permutation(i) = index + offset;
+      }
+      if (tmp_ranks(i) == next_biggest_rank) ++index;
+      if (last_pass) {
+        if (i + 1 == tmp_ranks.extent(0)) {
+          total() = index;
+        }
+        if (tmp_ranks(i) == next_biggest_rank) {
+          tmp_ranks(i) = -1;
+        }
+      }
+    });
+    auto host_total = Kokkos::create_mirror_view_and_copy(
+        Kokkos::HostSpace(), total);
+    auto count = host_total();
+    counts.push_back(count);
+    offset += count;
+  }
+}
+// clang-format on
+
+class Distributor
+{
+  public:
+    Distributor( MPI_Comm comm )
+        : _comm( comm )
+    {
+    }
+
+    template <typename T>
+    static T *notNullPtr( T *p )
+    {
+        return p ? p : reinterpret_cast<T *>( -1 );
+    }
+
+    template <typename View>
+    size_t createFromSends( View const &tmp )
+    {
+        static_assert(
+            std::is_same<typename View::non_const_value_type, int>::value, "" );
+        int comm_rank = -1;
+        MPI_Comm_rank( _comm, &comm_rank );
+
+        auto const n = tmp.size();
+        Kokkos::View<int *, Kokkos::HostSpace> destination_ranks(
+            "destination_ranks", n );
+        Kokkos::deep_copy( destination_ranks, tmp );
+        _permute = Kokkos::View<int *, Kokkos::HostSpace>( "permute", n );
+        std::vector<int> destinations;
+        extract_and_sort_ranks( destination_ranks, _permute, destinations,
+                                _dest_offsets, _dest_counts );
+
+        if ( _dest_counts.empty() )
+            _dest_offsets.push_back( 0 );
+        else
+            _dest_offsets.push_back( _dest_offsets.back() +
+                                     _dest_counts.back() );
+
+        {
+            int const reorder = 0; // Ranking may *not* be reordered
+            int const sources[1] = {comm_rank};
+            int const degrees[1] = {static_cast<int>( destinations.size() )};
+
+            MPI_Dist_graph_create(
+                _comm, 1, sources, degrees, notNullPtr( destinations.data() ),
+                MPI_UNWEIGHTED, MPI_INFO_NULL, reorder, &_comm_dist_graph );
+        }
+
+        int indegrees;
+        int outdegrees;
+        int weighted;
+        MPI_Dist_graph_neighbors_count( _comm_dist_graph, &indegrees,
+                                        &outdegrees, &weighted );
+        assert( weighted == 0 );
+        assert( outdegrees == (int)destinations.size() );
```

Comment:
Use `static_cast`

Reply:
Can do.  Just out of curiosity, why do you prefer the `static_cast`?